File: second.js
/*
 * blowing things up again - perhaps
 */

var fs = require('fs'),
    readline = require('readline'),
    matcher = /(https*:\/{2}.*?\/)\s*/i,
    accessLog = './access.log';

var rd = readline.createInterface({
    input: fs.createReadStream(accessLog),
    output: process.stdout,
    terminal: false
});

var sites = [],
    found = null;

function siteExists(site) {
    return sites.indexOf(site) > -1;
}

rd.on('line', function(line) {
    found = matcher.exec(line);
    if (found) {
        site = found[0];
        if (!siteExists(site)) {
            sites.push(site);
        }
    }
});

rd.on('close', function() {
    // console.log("Found all of those domains:\n");
    // sites.forEach(function(site) {
        // console.log("  *", site);
    // });
    console.log("Found", sites.length.toFixed(0), "URLs in the",
        accessLog, "file.");
});

File: second.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-

# will python blow things up?

import re

_URL_RE = re.compile("(https*:\\/{2}.*?\\/)\\s*", re.I)
_ACCESS_LOG = "access.log"


def get_access_log_contents():
    f = open(_ACCESS_LOG, 'r')
    for l in f.xreadlines():
        yield l
    f.close()

def run():
    sites = []
    for line in get_access_log_contents():
        match = _URL_RE.search(line) 
        if match:
            if match.group(0) not in sites:
                sites.append(match.group(0))
    # print "Found all of those domains:\n"
    # for site in sites:
        # print "  * %s" % site
    print "Found %i URLs in the %s file" % (len(sites), _ACCESS_LOG)


if __name__ == "__main__":
    run()
